go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -124,10 +124,13 @@ func isUnrecoverableError(err error) bool {
 	if err == nil {
 		return false
 	}
-	sqlErr, isSQLErr := err.(*mysql.SQLError)
+	sqlErr, isSQLErr := mysql.NewSQLErrorFromError(err).(*mysql.SQLError)
```

Comment:
I'm not sure how this is better? It should already be an SQLError. Won't isSQLErr always be true in this case?

Reply:
Answered in main comment